prow/plugins/config_test.go

```diff
@@ -1520,3 +1523,138 @@ orgA/repoB:
 		t.Error("expected unmarshal error but didn't get one")
 	}
 }
+
+func TestConfigMergingProperties(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name          string
+		makeMergeable func(*Configuration)
+	}{
+		{
+			name: "Plugins config",
+			makeMergeable: func(c *Configuration) {
+				*c = Configuration{Plugins: c.Plugins}
+			},
+		},
+	}
+
+	expectedProperties := []struct {
+		name         string
+		verification func(t *testing.T, fuzzedConfig *Configuration)
+	}{
+		{
+			name: "Merging into empty config always succeeds and makes the empty config equal to the one that was merged in",
+			verification: func(t *testing.T, fuzzedMergeableConfig *Configuration) {
+				newConfig := &Configuration{}
+				if err := newConfig.mergeFrom(fuzzedMergeableConfig); err != nil {
+					t.Fatalf("merging fuzzed mergeable config into empty config failed: %v", err)
+				}
+				if diff := cmp.Diff(newConfig, fuzzedMergeableConfig); diff != "" {
+					t.Errorf("after merging config into an empty config, the config that was merged into differs from the one we merged from:\n%s\n", diff)
+				}
+			},
+		},
+		{
+			name: "Merging empty config in always succeeds",
+			verification: func(t *testing.T, fuzzedMergeableConfig *Configuration) {
+				if err := fuzzedMergeableConfig.mergeFrom(&Configuration{}); err != nil {
+					t.Errorf("merging empty config in failed: %v", err)
+				}
+			},
+		},
+		{
+			name: "Merging a config into itself always fails",
+			verification: func(t *testing.T, fuzzedMergeableConfig *Configuration) {
+				if apiequality.Semantic.DeepEqual(fuzzedMergeableConfig, &Configuration{}) {
+					return
+				}
+
+				if err := fuzzedMergeableConfig.mergeFrom(fuzzedMergeableConfig); err == nil {
+					serialized, serializeErr := yaml.Marshal(fuzzedMergeableConfig)
+					if serializeErr != nil {
+						t.Fatalf("merging non-empty config into itself did not yield an error and serializing it afterwards failed: %v. Raw object: %+v", serializeErr, fuzzedMergeableConfig)
+					}
+					t.Errorf("merging a config into itself did not produce an error. Serialized config:\n%s", string(serialized))
+				}
+			},
+		},
+	}
+
+	seed := time.Now().UnixNano()
+	// Print the seed so failures can easily be reproduced
+	t.Logf("Seed: %d", seed)
+	fuzzer := fuzz.NewWithSeed(seed)
+
+	// Do not parallelize, the PRNG used by the fuzzer is not threadsafe
```

Comment:
Is this still true? It looks like _someone_ made this fuzzer thread-safe recently
https://github.com/google/gofuzz/pull/61

Reply:
Updated gofuzz and parallelized the test